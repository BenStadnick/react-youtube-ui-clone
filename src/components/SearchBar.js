import React, {useState} from 'react';
import Form from 'react-bootstrap/Form';

const SearchBar = ({onFormSubmit, defaultSearchTerm}) => {
  const [searchTerm, setSearchTerm] = useState(defaultSearchTerm);

  const handleChange = event => setSearchTerm(event.target.value);
  
  const handleSubmit = (event) => {
    onFormSubmit(searchTerm);
    event.preventDefault();
  }

  return (
    <div style={{ padding: '25px' }}>
      <Form onSubmit={handleSubmit}>
        <Form.Label>Search: </Form.Label>
        <Form.Control type='text' placeholder='Search...' value={searchTerm} onChange={handleChange} />
      </Form>
    </div>
  );
}

export default SearchBar;