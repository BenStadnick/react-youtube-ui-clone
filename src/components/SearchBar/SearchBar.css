.search_bar_container {
  text-align: center;
  padding-top: 25px;
}

.search_bar_input {
  width: 70%;
  display: inline-block;
}