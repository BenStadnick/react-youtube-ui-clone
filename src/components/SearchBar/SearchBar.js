import React, {useState} from 'react';
import {Form, InputGroup, Button} from 'react-bootstrap';
import './SearchBar.css'

const SearchBar = ({onFormSubmit, defaultSearchTerm}) => {
  const [searchTerm, setSearchTerm] = useState(defaultSearchTerm);

  const handleChange = event => setSearchTerm(event.target.value);
  
  const handleSubmit = (event) => {
    onFormSubmit(searchTerm);
    event.preventDefault();
  }

  return (
    <div className='search_bar_container'>
      <Form onSubmit={handleSubmit}  className='search_bar_input'>
        <InputGroup>
          <Form.Control type='text' placeholder='Search...' value={searchTerm} onChange={handleChange} />
          <InputGroup.Append>
            <Button  variant="outline-primary" type='submit'>Search</Button>
          </InputGroup.Append>
        </InputGroup>
      </Form>
    </div>
  );
}

export default SearchBar;