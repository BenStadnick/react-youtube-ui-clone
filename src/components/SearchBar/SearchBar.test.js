import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, mount} from 'enzyme';
import SearchBar from './SearchBar';


const mockOnFormSubmit = jest.fn( (x) => {return x;});

describe('SearchBar component', () => {
  it('did shallow render', () => {
    const wrapper = shallow(<SearchBar />);
    expect(wrapper).toMatchSnapshot();
  });

  it('shows default search term', () => {
    const defaultSearchTerm = 'test search term';
    const wrapper = mount(<SearchBar defaultSearchTerm={defaultSearchTerm} />);

    const textInput = wrapper.find('FormControl');
    expect(textInput.length).toBe(1); // single input exists
    expect(textInput.props().value).toBeDefined(); // has value
    expect(textInput.props().value).toEqual(defaultSearchTerm); // value is correct

    wrapper.unmount();
  });


  it('submits search on button form submint (button click or enter keypress)', () => {
    const defaultSearchTerm = 'test search term';
    const wrapper = mount(<SearchBar onFormSubmit={mockOnFormSubmit} defaultSearchTerm={defaultSearchTerm} />);
    
    const btn = wrapper.find('button');
    btn.simulate('submit');
    expect(mockOnFormSubmit).toHaveBeenCalled(); // called
    expect(mockOnFormSubmit.mock.results[0].value).toEqual(defaultSearchTerm); // called with correct value

    wrapper.unmount();
  });
});