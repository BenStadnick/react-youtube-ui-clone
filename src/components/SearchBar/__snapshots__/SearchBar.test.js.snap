// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component did shallow render 1`] = `
<div
  className="search_bar_container"
>
  <Form
    className="search_bar_input"
    inline={false}
    onSubmit={[Function]}
  >
    <InputGroup>
      <FormControl
        onChange={[Function]}
        placeholder="Search..."
        type="text"
      />
      <InputGroupAppend>
        <Button
          active={false}
          disabled={false}
          type="submit"
          variant="outline-primary"
        >
          Search
        </Button>
      </InputGroupAppend>
    </InputGroup>
  </Form>
</div>
`;
